src/app/auth/services/auth.service.ts

```diff
@@ -1,30 +1,82 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { Observable, BehaviorSubject } from "rxjs";
+import { SessionStorageService } from "./session-storage.service";
+import { User } from "@app/shared/models/user.model";
+import { tap } from "rxjs";
+
+import { Router } from "@angular/router";
+import { UserStoreService } from "@app/user/services/user-store.service";
+
+const TOKEN = "SESSION_TOKEN";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class AuthService {
-    login(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  private isAuthorized$$: BehaviorSubject<boolean>;
+  public isAuthorized$: Observable<boolean>;
```

Comment:
You can declare it by assigning the value BehaviorSubject.asObservable()
This way line 29 will be unnecessary.